packages/lib/server/repository/eventType.ts

```diff
@@ -0,0 +1,134 @@
+  Partial<
+    Omit<Prisma.EventTypeCreateInput, NotSupportedProps> & {
+      userId: PrismaEventType["userId"];
+      profileId: PrismaEventType["profileId"];
+      teamId: PrismaEventType["teamId"];
+      parentId: PrismaEventType["parentId"];
+      scheduleId: PrismaEventType["scheduleId"];
+    }
+  >,
+  "title" | "slug" | "length"
+>;
+
+export class EventTypeRepository {
+  static async create(data: IEventType) {
+    const {
+      userId,
+      profileId,
+      teamId,
+      parentId,
+      scheduleId,
+      bookingLimits,
+      recurringEvent,
+      metadata,
+      bookingFields,
+      durationLimits,
+      ...rest
+    } = data;
+    return await prisma.eventType.create({
+      data: {
+        ...rest,
+        ...(userId ? { owner: { connect: { id: userId } } } : null),
+        ...(profileId
+          ? {
+              profile: {
+                connect: {
+                  id: profileId,
+                },
+              },
+            }
+          : null),
+        ...(teamId ? { team: { connect: { id: teamId } } } : null),
+        ...(parentId ? { parent: { connect: { id: parentId } } } : null),
+        ...(scheduleId ? { schedule: { connect: { id: scheduleId } } } : null),
+        ...(metadata ? { metadata: metadata } : null),
+        ...(bookingLimits
+          ? {
+              bookingLimits,
+            }
+          : null),
+        ...(recurringEvent
+          ? {
+              recurringEvent,
+            }
+          : null),
+        ...(bookingFields
+          ? {
+              bookingFields,
+            }
+          : null),
+        ...(durationLimits
+          ? {
+              durationLimits,
+            }
+          : null),
+      },
+    });
+  }
+
+  static async findAllByUpId(
+    { upId }: { upId: string },
+    {
+      orderBy,
+      where = {},
+    }: { orderBy?: Prisma.EventTypeOrderByWithRelationInput[]; where?: Prisma.EventTypeWhereInput } = {}
+  ) {
+    if (!upId) return [];
+    const lookupTarget = ProfileRepository.getLookupTarget(upId);
+    const eventTypeWhere = {
+      ...(lookupTarget.type === LookupTarget.User
+        ? {
+            userId: lookupTarget.id,
+          }
+        : {
+            profileId: lookupTarget.id,
+          }),
+      ...where,
+    };
+    log.debug(
+      "findAllByUpId",
+      safeStringify({
+        upId,
+        orderBy,
+        argumentWhere: where,
+        where: eventTypeWhere,
+      })
+    );
+    return await prisma.eventType.findMany({
+      where: eventTypeWhere,
+      include: {
+        // TODO:  As required by getByViewHandler - Make it configurable
+        team: {
+          include: {
+            eventTypes: true,
+          },
+        },
+        hashedLink: true,
+        users: true,
+        children: {
+          include: {
+            users: true,
+          },
+        },
+        hosts: {
+          include: {
+            user: true,
+          },
+        },
```

Comment:
I don't want to be a downer here but this is causing leakage of team members undecoded avatar URIs and passwords. I'm getting payloads of about 30MB locally with production-like data.

Reply:
see #13549 @joeauyeung @hariombalhara